### PR TITLE
✨(helm) improve chart modularity and remove dependencies

### PR DESCRIPTION
**Pull request by wilbrdt**
## Purpose

Ralph Helm chart still suffers from its legacy (mimic of Arnold's tray).

### Added

- A NOTE.txt file with instructions for chart users
- A ConfigMap template for Ralph configuration
- A ConfigMap for logging configuration
- A secret Manifest for helping user configure Ralph

### Changed

- Environment variables are now provided through a ConfigMap and a Secret
- Improve Ingress configuration
- Improved values based on Helm chart template file 

### Removed

- Dependencies to mongodb and clickhouse (managed outside of chart scope)
- Secrets template and vault values (managed outside of chart scope)

### CODE REVIEW

File: src/helm/ralph/Chart.yaml [hunk 1]
```diff
@@ -1,17 +1,8 @@
 apiVersion: v2
 
 name: ralph
+description: Ralph LRS
```
Comment (jmaupetit) [inline src/helm/ralph/Chart.yaml:4]:
```suggestion
description: Ralph, the ultimate Learning Record Store (and more!) for your learning analytics 
```

File: src/helm/manifests/ralph-env-cm.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+---
```
Comment (jmaupetit) [inline src/helm/manifests/ralph-env-cm.yaml:1]: This should be a distributed template instead, with all configurations overridable by defining the corresponding Value.

**Review by jmaupetit — COMMENTED** [review]
Wow, amazing :tada:

File: src/helm/manifests/auth-demo.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    "username": "admin",
+    "hash": "$2b$12$JFK.YCdbUWD2rS94fT4.m.KC/fIMzUMPMtjaD4t3t1iAfqki3ZPOq",
+    "agent": {
+    "mbox": "mailto:admin@example.com",
+    "objectType": "Agent",
+    "name": null
+    },
+    "scopes": ["example_scope"]
+}
+]
```
Comment (jmaupetit) [inline src/helm/manifests/auth-demo.json:12]:
```suggestion
[
  {
    "username": "admin",
    "hash": "$2b$12$JFK.YCdbUWD2rS94fT4.m.KC/fIMzUMPMtjaD4t3t1iAfqki3ZPOq",
    "agent": {
      "mbox": "mailto:admin@example.com",
      "objectType": "Agent",
      "name": null
    },
    "scopes": [
      "example_scope"
    ]
  }
]
```

File: src/helm/ralph/templates/NOTES.txt [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+{{- if .Values.ingress.enabled }}
+{{- range $host := .Values.ingress.hosts }}
+  {{- range .paths }}
+  http{{ if $.Values.ingress.tls }}s{{ end }}://{{ $host.host }}{{ .path }}
+  {{- end }}
+{{- end }}
+{{- else if contains "NodePort" .Values.service.type }}
+  export NODE_PORT=$(kubectl get --namespace {{ .Release.Namespace }} -o jsonpath="{.spec.ports[0].nodePort}" services {{ include "ralph.fullname" . }})
+  export NODE_IP=$(kubectl get nodes --namespace {{ .Release.Namespace }} -o jsonpath="{.items[0].status.addresses[0].address}")
+  echo http://$NODE_IP:$NODE_PORT
```
Comment (jmaupetit) [inline src/helm/ralph/templates/NOTES.txt:17]: That's odd defaults.

File: src/helm/ralph/values.yaml [hunk 5]
```diff
@@ -122,17 +210,5 @@ lrs:
 #         gunzip |
 #         ralph extract -p gelf |
 #         ralph write \
-#         --backend es \
-#         --es-client-options ca_certs=/usr/local/share/ca-certificates/es-cluster.pem"
+#         --backend es
 cronjobs: []
```
Comment (jmaupetit) [inline src/helm/ralph/values.yaml:214]: For consistency, maybe put `cronjobs` commented example below?

**Review by jmaupetit — COMMENTED** [review]
That's a wonderful job! :sparkles: 

Maybe I would add our expectations about Ralph's env secret file in the Helm Chart README. WDYT?

File: src/helm/ralph/templates/cronjob.yaml [hunk 6]
```diff
@@ -15,69 +15,80 @@ spec:
   successfulJobsHistoryLimit: 2
   concurrencyPolicy: Forbid
   failedJobsHistoryLimit: 1
-  suspend: false
+  suspend: False
```
Comment (claudusd) [inline src/helm/ralph/templates/cronjob.yaml:18]: Should be a variable.

File: src/helm/manifests/ralph-env-secret.yaml [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+
+  ## LRS HTTP backend
+  # RALPH_BACKENDS__DATA__LRS__BASE_URL: "http://ralph:secret@0.0.0.0:8100/"
+  # RALPH_BACKENDS__DATA__LRS__USERNAME: "ralph"
+  # RALPH_BACKENDS__DATA__LRS__PASSWORD: "secret"
+  # RALPH_BACKENDS__DATA__LRS__HEADERS__X_EXPERIENCE_API_VERSION: "1.0.3"
+  # RALPH_BACKENDS__DATA__LRS__HEADERS__CONTENT_TYPE: "application/json"
+  # RALPH_BACKENDS__DATA__LRS__STATUS_ENDPOINT: "/__heartbeat__"
+  # RALPH_BACKENDS__DATA__LRS__STATEMENTS_ENDPOINT: "/xAPI/statements"
+---
```
Comment (claudusd) [inline src/helm/manifests/ralph-env-secret.yaml:139]: For me except the key ending by ``__USERNAME``,  ``__PASSWORD``, ``__ACCESS_KEY_ID`` or ``__SECRET_ACCESS_KEY`` should be in a configmap.

File: src/helm/ralph/templates/ingress.yaml [hunk 8]
```diff
@@ -1,44 +1,61 @@
...
+  {{- $_ := set .Values.ingress.annotations "kubernetes.io/ingress.class" .Values.ingress.className}}
+  {{- end }}
+{{- end }}
+{{- if semverCompare ">=1.19-0" .Capabilities.KubeVersion.GitVersion -}}
 apiVersion: networking.k8s.io/v1
+{{- else if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
+apiVersion: networking.k8s.io/v1beta1
+{{- else -}}
+apiVersion: extensions/v1beta1
+{{- end }}
```
Comment (claudusd) [inline src/helm/ralph/templates/ingress.yaml:15]: @jmaupetit we have already talk about support deprecated version. But I don't remember your decision.

**Review by claudusd — CHANGES_REQUESTED** [review]

File: src/helm/ralph/templates/cronjob.yaml (hunk 6 above)
Comment (wilbrdt) [reply src/helm/ralph/templates/cronjob.yaml:18]: Good point, thanks! Added a variable

File: src/helm/manifests/ralph-env-secret.yaml (hunk 7 above)
Comment (wilbrdt) [reply src/helm/manifests/ralph-env-secret.yaml:139]: Good remark. I created a ConfigMap manifest for non-secret environment variables.
I really don't want to add them as variable in the Helm chart :grin:

File: src/helm/ralph/templates/ingress.yaml (hunk 8 above)
Comment (wilbrdt) [reply src/helm/ralph/templates/ingress.yaml:15]: You would advise against supporting depredated k8s version?

Comment (claudusd) [reply src/helm/ralph/templates/ingress.yaml:15]: Yes, but it's my opinion.

**Review by claudusd — APPROVED** [review]
I let you decide about deprecated version for K8S.

File: src/helm/ralph/templates/ingress.yaml (hunk 8 above)
Comment (wilbrdt) [reply src/helm/ralph/templates/ingress.yaml:15]: Leaving as is for now!

**Pull request merged**